File: backend/src/main/java/com/dungeon/heotaehoon/entity/Instructor.java
package com.dungeon.heotaehoon.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "instructors")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Instructor {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(nullable = false, unique = true, length = 50)
    private String username;

    @Column(nullable = false, length = 255)
    private String password;

    @Column(nullable = false, length = 100)
    private String name;

    @Builder.Default
    @Column(nullable = false)
    private Integer level = 1;

    @Builder.Default
    @Column(nullable = false)
    private Integer exp = 0;

    @Column(length = 500)
    private String currentTitle;

    @Builder.Default
    @Column(name = "rage_gauge", nullable = false)
    private Integer rageGauge = 0;

    @Builder.Default
    @Column(name = "is_evolved", nullable = false)
    private Boolean isEvolved = false;

    @Column(name = "evolution_stage", length = 50)
    private String evolutionStage = "normal";

    @Column(name = "voice_model_url")
    private String voiceModelUrl;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/lib/models/instructor.dart
class Instructor {
  final String id;
  final String name;
  final String username;
  final int level;
  final int exp;
  final int rageGauge;

  Instructor({
    required this.id,
    required this.name,
    required this.username,
    required this.level,
    required this.exp,
    required this.rageGauge,
  });

  factory Instructor.fromJson(Map<String, dynamic> json) {
    return Instructor(
      id: json['id'],
      name: json['name'],
      username: json['username'],
      level: json['level'],
      exp: json['exp'],
      rageGauge: json['rageGauge'],
    );
  }
}
